firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/AnalyticsDeferredComponents.java

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics;
+
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.APP_EXCEPTION_CALLBACK_TIMEOUT_MS;
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.FIREBASE_CRASHLYTICS_ANALYTICS_ORIGIN;
+import static com.google.firebase.crashlytics.FirebaseCrashlytics.LEGACY_CRASH_ANALYTICS_ORIGIN;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+import com.google.firebase.analytics.connector.AnalyticsConnector;
+import com.google.firebase.crashlytics.internal.Logger;
+import com.google.firebase.crashlytics.internal.analytics.AnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.BlockingAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.BreadcrumbAnalyticsEventReceiver;
+import com.google.firebase.crashlytics.internal.analytics.CrashlyticsOriginAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.analytics.UnavailableAnalyticsEventLogger;
+import com.google.firebase.crashlytics.internal.breadcrumbs.BreadcrumbSource;
+import com.google.firebase.crashlytics.internal.breadcrumbs.DisabledBreadcrumbSource;
+import com.google.firebase.inject.Deferred;
+import java.util.concurrent.TimeUnit;
+
+/** @hide */
+public class AnalyticsDeferredComponents {
```

Comment:
Since both objects that depend on the deferred objects are interfaces in Crashlytics , I've addressed your comment as follows:

1. Created new "proxy" implementations of the interfaces, these implementations are inner classes to the `AnalyticsDeferredComponent` object. 
2. These proxy objects simply channel any calls to the actual implementing object in the `AnalyticsDeferredComponent` object. 
3. The `AnalyticsDeferredComponent` by default creates 2 objects which are the default implementations Crashlytics use when there's no `AnalyticsConnector` present.
4. When the `AnalyticsConnector` becomes available, we switch the default implementation objects with new ones that use Analytics.

This is transparent to any other objects using `BreadcrumbSource` and `AnalyticsEventLogger` since the proxy implementations will switch to the right objects cleanly.

Reply:
PTAL